clang/utils/TableGen/ClangDiagnosticsEmitter.cpp

```diff
@@ -1553,15 +1526,13 @@ static void emitDiagSubGroups(std::map<std::string, GroupInfo> &DiagsInGroup,
                               RecordVec &GroupsInPedantic, raw_ostream &OS) {
   OS << "static const int16_t DiagSubGroups[] = {\n"
      << "  /* Empty */ -1,\n";
-  for (auto const &I : DiagsInGroup) {
-    const bool IsPedantic = I.first == "pedantic";
-
-    const std::vector<std::string> &SubGroups = I.second.SubGroups;
+  for (auto const &[Name, Group] : DiagsInGroup) {
```

Comment:
Same question here.

Reply:
Here, the & is so prevent making a copy, but I guess what you are saying is that it won't be a copy but a reference even without the &?

Reply:
Ah, actually, i think it does make sense here, this is a string/group-info pair, so that is worth reference here.

Reply:
If I remove the &, I get:

>warning: loop variable '[Name, Group]' creates a copy from type 'std::pair<const std::basic_string<char>, (anonymous > namespace)::GroupInfo> const' [-Wrange-loop-construct

So the & here seems required to prevent copies.

Reply:
Yep, agree.  This is a big enough one to use a reference.  In general, the `reference vs copy` when there are reasonably trivial copy operations, is roughly 2x ptr size.  At this point, I think we're beyond that line (plus have sufficiently non-trivial copy operations), so a reference is valuable here.

I was perhaps a touch comment-happy/didn't look as closely on this one after seeing the previous one.

Reply:
So I guess we are good with resolving this?